lib/src/git_backend.rs

```diff
@@ -2359,15 +2359,17 @@ mod tests {
     // UserSettings type. testutils returns jj_lib (2)'s UserSettings, whereas
     // our UserSettings type comes from jj_lib (1).
     fn user_settings() -> UserSettings {
```

Comment:
nit: Can you leave a comment why `git.write-change-id-header` is turned off by default in tests? It seems odd that `user_settings()` with no suffix does more things than `user_settings_with_change_id()`.

Reply:
I've turned it on instead :)